src/main/java/io/anserini/collection/CommonCrawlWetCollection.java

```diff
@@ -106,6 +108,96 @@ public static Document readNextWarcRecord(DataInputStream in)
       return retRecord;
     }
 
+    /**
+     * Performs the actual heavy lifting of reading in the next WARC record.
+     *
+     * @param in the data input stream
+     * @param headerBuffer a blank string buffer to contain the WARC header
+     * @return the content bytes (with the headerBuffer populated)
+     * @throws IOException if error encountered reading from stream
+     */
+    protected static byte[] readNextRecord(DataInputStream in, StringBuilder headerBuffer, String headerEndKey) throws IOException {
+      if (in == null || headerBuffer == null) {
+        throw new NoSuchElementException();
+      }
+
+      String line = null;
+      boolean foundMark = false;
+      boolean inHeader = true;
+      byte[] retContent = null;
+
+      // cannot be using a buffered reader here!!!!
```

Comment:
Can you clarify these comments and below?

Reply:
I'll remove the comment about buffered reader here. It should belong to `readLineFromInputStream` and was well described in that function.